src/d/actor/d_a_obj_shelf.cpp

```diff
@@ -4,133 +4,288 @@
 //
 
 #include "d/actor/d_a_obj_shelf.h"
+#include "d/res/res_otana.h"
 #include "d/d_procname.h"
+#include "d/d_com_inf_game.h"
+#include "d/actor/d_a_npc_ba1.h"
+#include "m_Do/m_Do_mtx.h"
+
+#include "weak_data_1811.h" // IWYU pragma: keep
+
+Mtx daObjShelf::Act_c::M_tmp_mtx;
+
+namespace daObjShelf {
+    namespace {
+        struct Attr_c {
+            /* 0x00 */ short mVibDuration;
+            /* 0x02 */ short mVibYSpeed;
+            /* 0x04 */ float mVibYMagnitude;
+            /* 0x08 */ short mVibXSpeed;
+            /* 0x0a */ short mVibZSpeed;
+            /* 0x0c */ float mVibXMagnitude;
+            /* 0x10 */ float mVibZMagnitude;
+            /* 0x14 */ float mRotAccel;
+            /* 0x18 */ float mRotDecay;
+            /* 0x1c */ float mBounceFactor;
+            /* 0x20 */ u8 mBounceNum;
+            /* 0x21 */ u8 mRotWaitDuration;
+            /* 0x22 */ short m22;
```

Comment:
this is unused?

Reply:
It's probably just alignment padding. mRotWaitDuration ends at offset 0x22, but mInitSpeed2 is a float, which has alignment 4. So the compiler places mInitSpeed2 at offset 0x24 for alignment, and these two bytes in here are just padding. You can remove this line.